scripts/linum_merge_slices_into_folders.py

```
@@ -0,0 +1,127 @@
+
+    :return: None
+    """
+    # Create new folders per z slice
+    for folder in tqdm(folders, desc="Copying data from old folders to new z slice folders", unit="folder"):
+        z_slice = folder.name.split("_")[-1]
+        new_folder = tiles_directory / f"{z_slice}"
+        new_folder.mkdir(exist_ok=True)
+        # Copy all contents from the old folder to the new folder, keeping the structure but removing the z slice from the folder name
+        for item in folder.iterdir():
+            if item.is_file():
+                new_sub_folder = new_folder / folder.name
+                new_sub_folder.mkdir(exist_ok=True)
+                shutil.copy(item, new_sub_folder / item.name)
+
+
+def check_files(tiles_directory: Path, old_folders: list) -> None:
+    """
+    Check if all files were moved correctly by comparing the new folders with the old ones.
+
+    :param tiles_directory: Path to the directory containing the tiles.
+    :type tiles_directory: Path
+    :param old_folders: List of old folders containing the tiles.
+    :type old_folders: list[Path]
+    """
+    print("Checking if all files were moved correctly...")
+    equal = True
+    for folder in old_folders:
+        z_slice = folder.name.split("_")[-1]
+        new_folder = tiles_directory / f"{z_slice}"
+        new_sub_folder = new_folder / folder.name
+        # Check if the new sub folder exists
+        if not new_sub_folder.exists():
+            print(f"New sub folder {new_sub_folder} does not exist")
+            equal = False
+            continue
+        # Check if the files in the old folder are in the new sub folder
+        for item in folder.iterdir():
+            if item.is_file():
+                if not (new_sub_folder / item.name).exists():
+                    print(f"File {item.name} from {folder} is not in {new_sub_folder}")
+                    equal = False
+
+    if not equal:
+        print("Not all files were moved correctly, check the output above.")
```

We could raise an error and exit here.

---

See comment above.